Pilot/tests/Test_PilotLogger.py

```diff
@@ -1,60 +1,115 @@
     try:
-      os.remove(fr)
+      os.remove(self.uuidFile)
     except OSError:
       pass
 
+  def test_success(self):
+    for status in self.logger.STATUSES:
+      self.assertTrue(self.logger._isCorrectStatus(status))
+
+  def test_failure(self):
+    self.assertFalse(self.logger._isCorrectStatus('mamma Mia'))
 
-def test_success(rmFiles):
-  getUniqueIDAndSaveToFile(testFile)
-  uuid = getPilotUUIDFromFile(testFile)
-  assert uuid
+  def test_failureEmpty(self):
+    self.assertFalse(self.logger._isCorrectStatus(''))
 
 
-def test_failureBadFile(rmFiles):
-  getUniqueIDAndSaveToFile(testFile)
-  uuid = getPilotUUIDFromFile(badFile)
-  assert not uuid
+class TestPilotLogger_init(unittest.TestCase):
 
+  def setUp(self):
+    self.uuidFile = 'PilotUUID'
 
-def test_failureNonExistent(rmFiles):
-  uuid = getPilotUUIDFromFile(nonExistentFile)
-  assert not uuid
+  def tearDown(self):
+    try:
+      os.remove(self.uuidFile)
+    except OSError:
+      pass
+
+  def test_DefaultCtrNonJsonFile(self):
+    logger = PilotLogger()
+    self.assertEqual(logger.params['LoggingType'], 'LOCAL_FILE')
+    self.assertEqual(logger.params['LocalOutputFile'], 'myLocalQueueOfMessages')
+    self.assertEqual(logger.params['FileWithID'], 'PilotUUID')
+
+
+class TestPilotLogger_addMissingConfiguration(unittest.TestCase):
+
+  def setUp(self):
+    self.uuidFile = 'PilotUUID'
+
+  def tearDown(self):
+    try:
+      os.remove(self.uuidFile)
+    except OSError:
+      pass
 
+  def test_success(self):
+    config = {'LoggingType': 'MQ',
+              'LocalOutputFile': 'blabla', 'FileWithID': 'myUUUID'}
+    res = addMissingConfiguration(config)
+    self.assertEqual(res, config)
 
-def test_PilotLogger_success(rmFiles):
-  logger = PilotLogger()
-  for status in logger.STATUSES:
-    assert logger._isCorrectStatus(status) is True
+  def test_emptyConfig(self):
+    self.assertEqual(addMissingConfiguration(None),
+                     {'LoggingType': 'LOCAL_FILE',
+                      'LocalOutputFile': 'myLocalQueueOfMessages',
+                      'FileWithID': 'PilotUUID'})
 
 
-def test_failure(rmFiles):
-  logger = PilotLogger()
-  assert logger._isCorrectStatus('mamma Mia') is False
+class TestPilotLogger_sendMessage(unittest.TestCase):
+  pass
+  # here some mocks needed
 
 
-def test_failureEmpty(rmFiles):
-  logger = PilotLogger()
-  assert logger._isCorrectStatus('') is False
+if __name__ == '__main__':
+  suite = unittest.defaultTestLoader.loadTestsFromTestCase(TestGetPilotUUIDFromFile)
+  suite.addTest(unittest.defaultTestLoader.loadTestsFromTestCase(TestPilotLogger_isCorrectStatus))
+  suite.addTest(unittest.defaultTestLoader.loadTestsFromTestCase(TestPilotLogger_init))
+  suite.addTest(unittest.defaultTestLoader.loadTestsFromTestCase(TestPilotLogger_addMissingConfiguration))
+  suite.addTest(unittest.defaultTestLoader.loadTestsFromTestCase(TestPilotLogger_sendMessage))
+  testResult = unittest.TextTestRunner(verbosity=2).run(suite)
```

Comment:
Please, as last lines of these integration tests (all those in Pilot/tests) that are coded using python `unittest`, use:

```
  sys.exit(not testResult.wasSuccessful())
```

This way it will exit with return code !=0 if not successful (it's useful in Jenkins and other CI tools).

This is true for this file and the others in this dir too.